### PR TITLE
Add IBM Watson STT Json import support.

### PR DESCRIPTION
**Pull request by AndrewDAnderson**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
No.

**Describe what the PR does**    
Adds initial support for IBM Watson TTS Json files. Working well in my limited tests, but a few things could probably be refactored for clarity. Creating EntityRanges is done from within the adapter itself, rather than being handed off to the common function for that purpose...

**State whether the PR is ready for review or whether it needs extra work**    
Ready to be reviewed.

**Additional context**    
N/A

### CODE REVIEW

File: src/lib/Util/adapters/ibm/index.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        const startTime = result.alternatives[0].timestamps[z][1];
+        const endTime = result.alternatives[0].timestamps[z][2];
+        const wordConfidence = result.alternatives[0].hasOwnProperty('word_confidence') ?
+          result.alternatives[0].word_confidence[z][1] : 100;
+
+        if (newSentence === true) {
+          punctuatedWord = capitalizeFirstLetter(word);
+          firstWordOfSentence = word;
+          newSentence = false;
+          data.speaker = 'TBC ' + topLvlResult.speaker_labels[z].speaker;
```
Comment (pietrop) [inline src/lib/Util/adapters/ibm/index.js]: IBM gives you speaker labels, and I've seen you are using that in the adapter, which is great
Only minor thing, I'd change the `TBC` to `Speaker`?

```suggestion
           data.speaker = 'Speaker ' + topLvlResult.speaker_labels[z].speaker;
```

As we have been (informally) using `TBC` as a place holder when the STT service doesn't return any speaker diarization info, and/or when it has not been incorporated in the adapter yet.

**Comment by pietrop** [comment]
That's great, thanks @AndrewDAnderson 

I guess you meant to write STT (Speech To Text) and not TTS (Text To Speech) in the title of the PR (?).

Looks good, and thanks for adding tests!

A few comments 

1. As you mentioned Creating EntityRanges could be done with the shared helper function, unless there are any advantages in doing that within the IBM adapter?

2. Made a comment/suggestion re-speaker label names - see inline.

3. It might be good to have more meaningful names for `x`, `y`,`z`, I do realised they are just loop counters, for example `topLevelCounter`,`wordLevelCounter` and `sentenceLevelCounter` (?)

_We'll have another look and see if there's anything else, thanks_

File: src/lib/Util/adapters/ibm/index.js (hunk 1 above)
Comment (AndrewDAnderson) [reply src/lib/Util/adapters/ibm/index.js]: Sounds great. I had originally had it as "Speaker" but didn't know if "TBC" had any... meaning and was trying to make this adapter 'fit in' as much as possible. ;-)

**Comment by AndrewDAnderson** [comment]
> I guess you meant to write STT (Speech To Text) and not TTS (Text To Speech) in the title of the PR (?).

Yes. Note to self: do not make public commits at 3am.

> 1. As you mentioned Creating EntityRanges could be done with the shared helper function, unless there are any advantages in doing that within the IBM adapter?

Honestly, I didn't investigate the helper function at all - I had already started writing an adapter for a project and just tweaked it a little for use here. I do have questions about why there is so much duplication of data - but I will ask those in place.

> 2. Made a comment/suggestion re-speaker label names - see inline.

Responded inline. Good suggestion.

> 3. It might be good to have more meaningful names for `x`, `y`,`z`, I do realised they are just loop counters, for example `topLevelCounter`,`wordLevelCounter` and `sentenceLevelCounter` (?)

Sounds good. This was just a quick attempt to get it *functional* - and I'm sure others would like basic IBM support, so I was pushing it out the door with plans of cleaning things up later.

**Comment by pietrop** [comment]
Hi @AndrewDAnderson , 
Just circling back to see if you are interested in re-basing this, and address some of the suggestions mentioned above so that we can merge it into master?

We just added support for Speechmatics https://github.com/bbc/react-transcript-editor/pull/94 and thought might be good to wrap up the IBM one as well.

Let me know

**Pull request merged**